Source/DiabloUI/art.cpp

```diff
@@ -40,7 +40,7 @@ bool LoadPcxPixelsAndPalette(HANDLE handle, int width, int height, std::uint8_t
 
 	// We read 1 extra byte because it delimits the palette.
 	const unsigned readSize = pixelDataSize + (has256ColorPalette ? PcxPaletteSize : 0);
-	std::unique_ptr<BYTE[]> fileBuffer = std::make_unique<BYTE[]>(readSize);
+	std::unique_ptr<BYTE[]> fileBuffer { new BYTE[readSize] };

```

Comment:
Why though? What is the cost associated with the zero initialization? Is it enough to drive the code into an unusual initialization scheme? Every `unique_ptr` is initialized using `make_unique` and suddenly there are now cases where it isn't.

Is this really worth it?

Reply:
Yes, sometimes there is a cost associated with zero initialized. We're filling buffers here, it's fairly low-level code.

Reply:
If you'd like to make it nicer, you could backport `make_unique_for_overwrite` from C++20

Reply:
There is an example implementation of it here:
https://en.cppreference.com/w/cpp/memory/unique_ptr/make_unique